src/Compilers/CSharp/Test/Symbol/Symbols/Source/FileModifierTests.cs

```diff
@@ -2,48 +2,2433 @@
+                {
+                    public static void M() { }
+                }
+            }
+            """;
+
+        var compilation = CreateCompilation(new[] { source1, source2, source3 });
+        compilation.VerifyDiagnostics();
+
+        var classOuter = compilation.GetMember<NamedTypeSymbol>("Outer");
+        var cs = classOuter.GetMembers("C");
+        Assert.Equal(3, cs.Length);
+        Assert.True(cs[0] is SourceMemberContainerTypeSymbol { IsFile: true });
+        Assert.True(cs[1] is SourceMemberContainerTypeSymbol { IsFile: true });
+        Assert.True(cs[2] is SourceMemberContainerTypeSymbol { IsFile: false });
+    }
+
+    [Fact]
+    public void Duplication_09()
+    {
+        var source1 = """
+            namespace NS
+            {
+                file class C
+                {
+                    public static void M() { }
+                }
+            }
+            """;
+
+        var source2 = """
+            namespace NS
+            {
+                file class C
+                {
+                    public static void M() { }
+                }
+            }
+            """;
+
+        var source3 = """
+            namespace NS
+            {
+                public class C
+                {
+                    public static void M() { }
+                }
+            }
+            """;
+
+        var compilation = CreateCompilation(new[] { source1, source2, source3 });
+        compilation.VerifyDiagnostics();
+
+        var namespaceNS = compilation.GetMember<NamespaceSymbol>("NS");
+        var cs = namespaceNS.GetMembers("C");
+        Assert.Equal(3, cs.Length);
+        Assert.True(cs[0] is SourceMemberContainerTypeSymbol { IsFile: true });
+        Assert.True(cs[1] is SourceMemberContainerTypeSymbol { IsFile: true });
+        Assert.True(cs[2] is SourceMemberContainerTypeSymbol { IsFile: false });
+    }
+
+    [Theory]
+    [InlineData("file", "file")]
+    [InlineData("file", "")]
+    [InlineData("", "file")]
+    public void Duplication_10(string firstFileModifier, string secondFileModifier)
+    {
+        var source1 = $$"""
+            using System;
+
+            partial class Program
+            {
+                {{firstFileModifier}} class C
+                {
+                    public static void M()
+                    {
+                        Console.Write(1);
+                    }
+                }
+            }
+            """;
+
+        var source2 = $$"""
+            using System;
+
+            partial class Program
+            {
+                {{secondFileModifier}} class C
+                {
+                    public static void M()
+                    {
+                        Console.Write(2);
+                    }
+                }
+            }
+            """;
+
+        var main = """
+            partial class Program
+            {
+                static void Main()
+                {
+                    Program.C.M();
```

Comment:
>Program. [](http://example.com/codeflow?start=20&length=8)

The qualifier isn't needed is it? Just confirming.

Reply:
The meaning of the program isn't changed, but the method of binding is changed. Here we lookup the symbols directly in the Program type rather than in a binder which contains the members of Program. I think I added this to address the issue discussed in https://github.com/dotnet/roslyn/pull/60977#issuecomment-1126604537.